test/distrib/cpp/run_distrib_test.sh

```diff
@@ -30,9 +30,13 @@
 
 set -ex
 
-git clone --recursive $EXTERNAL_GIT_ROOT
+git clone $EXTERNAL_GIT_ROOT
 cd grpc
 
+# clone submodules
+git submodule | awk -v EXTERNAL_GIT_ROOT=$EXTERNAL_GIT_ROOT/ '{ system("git \
```

Comment:
please take a look at the git submodule foreach suggestion below and decide which one is more readable. Also elaborate the comment a bit.